src/appShell/App/Container.tsx

```diff
@@ -10,31 +11,86 @@ interface IContainerProps {
     children: React.ReactNode;
 }
 
-export default class Container extends React.Component<IContainerProps, void> {
+interface IContainerState {
+    showModal: boolean;
+    ieVersion: string;
+}
+
+export default class Container extends React.Component<IContainerProps, IContainerState> {
+
+    constructor() {
+        super();
+
+        this.state = {showModal: false, ieVersion: ''};
+        this.handleCloseClick = this.handleCloseClick.bind(this);
+    }
 
     static contextTypes = {
         router: React.PropTypes.object,
     };
 
     context: {router: any};
 
+    findIEVersion(userAgent:string, versionIndex: number, localStorage:boolean = false) {
```

Comment:
i think this method should have a more general name.  like, handleUnsupportedBrowsers.    also, since you are using localStorage within this method, you might as well internalize the check for browser setting, instead of passing it in.  in other words, this method is NOT pure in the functional-programming sense, so you might as well enjoy it, so to speak.  let me know if that needs more explanation.